File: network_supervision_port.go
package main

type networkSupervisionPort interface {
    /**
     * Supervision agents on devices check connectivity between them, and
     * may raise alerts when a network interface link with another Device seems to be down.
     * This callback will be called whenever this occurs.
     *
     * @param sourceDeviceName the Device on which the agent is running
     * @param targetDeviceName the Device that the agent failed to reach
     */
    onNetworkInterfaceAlert(sourceDeviceName string, targetDeviceName string)
}

File: src/main/java/com/orange/craftsmanship/alarming/domain/NetworkSupervisionPort.java
package com.orange.craftsmanship.alarming.domain;

public interface NetworkSupervisionPort {
    /**
     * Supervision agents on devices check connectivity between them, and
     * may raise alerts when a network interface link with another device seems to be down.
     * This callback will be called whenever this occurs.
     *
     * @param sourceDeviceName the device on which the agent is running
     * @param targetDeviceName the device that the agent failed to reach
     */
    void onNetworkInterfaceAlert(String sourceDeviceName, String targetDeviceName);
}
